server/controllers/auth/login.js

```diff
@@ -0,0 +1,39 @@
+const bcrypt = require('bcrypt');
+const { sign } = require('jsonwebtoken');
+
+const loginValidation = require('../../utils/validation/loginValidation');
+const applicant = require('../../database/models/applicant');
+require('env2')('config.env');
+
+const SECRET = process.env.SECRET_KEY;
+
+const login = (req, res) => {
+  const { email, password } = req.body;
+  loginValidation(email, password)
+    .then((valid) => {
+      if (!valid)
+        res.status(400).json({ error: 'Please fill in all required fields' });
+      else {
+        applicant
+          .findOne({ email })
+          .then((data) => {
+            bcrypt.compare(password, data.password, (err, result) => {
+              if (!result) res.status(400).json({ error: 'Wrong Password' });
+              else {
+                const userToken = { userId: data.id };
+                const cookie = sign(userToken, SECRET);
+                res.cookie('applicant', cookie, { httpOnly: true }).json(data);
```

Comment:
I think you should define a clear  structure for success and for failure something like this
with success send an object with this shape
```js
.json({
   data: data
  //or could be just data using es6
   data
})
```

and for errors the same as you are doing right now

```js
.json({
   error: err
})
```
@Asmaathabet @mFlaifel @ha77an @alidah0 
what do you think?

Reply:
I totally agree with this, it should be modified to be more clear , and to display the type of returned data/error and its shape.